src/command_modules/azure-cli-interactive/azclishell/telemetry.py

```diff
@@ -19,27 +22,53 @@
 INSTRUMENTATION_KEY = '762871d5-45a2-4d67-bf47-e396caf53d9d'
 
 
-def my_context(tel_client):
-    """ context for the application """
-    tel_client.context.application.id = 'Azure CLI Shell'
-    tel_client.context.application.ver = __version__
-    tel_client.context.user.id = Profile().get_installation_id()
-    tel_client.context.instrumentation_key = INSTRUMENTATION_KEY
+class TelThread(threading.Thread):
+    """ telemetry thread for exiting """
+    def __init__(self, threadfunc):
+        threading.Thread.__init__(self)
+        self.threadfunc = threadfunc
+
+    def run(self):
+        """ pushes the function out """
+        try:
+            self.threadfunc()
+        except KeyboardInterrupt:
+            pass
 
 
 class Telemetry(TelemetryClient):
     """ base telemetry sessions """
 
-    start_time = None
-    end_time = None
+    def __init__(self, *args, **kwargs):
+        super(Telemetry, self).__init__(*args, **kwargs)
+        self.start_time = None
+        self.end_time = None
+        enable(args[0])  # is the Instrument Key
+        self.add_context()
+
+    def add_context(self):
```

Comment:
If the only consumer of this method is the constructor and the method have only four lines, why not accomplish it in the constructor?